client/src/epics/revisions.js

```diff
@@ -164,169 +133,67 @@ export const fetchRevision = ( action$, store ) => (
-						} )
-						// If the query is no longer valid, cancel the request.
-						.takeUntil( action$.ofType( 'REVISIONS_NOT_READY' ) )
-						// If the user is no longer in the query, cancel the request.
-						.takeUntil( action$.ofType( 'QUERY_USER_CHANGE' ).filter( a => !a.users.includes( user ) ) )
-						// If the wiki changes, cancel the request.
-						.takeUntil( action$.ofType( 'QUERY_WIKI_CHANGE' ).filter( a => a.wiki !== wiki ) )
-						// If the start date changes, cancel the request.
-						.takeUntil( action$.ofType( 'QUERY_START_DATE_CHANGE' ).filter( a => a.startDate !== startDate ) )
-						// If the end date changes, cancel the request.
-						.takeUntil( action$.ofType( 'QUERY_END_DATE_CHANGE' ).filter( a => a.endDate !== endDate ) );
-				} );
-
-			return Observable.forkJoin( requests.toArray() )
-				.flatMap( ( data ) => {
-
-					if ( data.length === 0 ) {
-						return Observable.of( addRevisions( new OrderedMap() ) );
-					}
-
-					const revisions = data.reduce( ( map, item ) => {
-						return map.merge( item.revisions );
-					}, new OrderedMap() );
-					const cont = data.reduce( ( map, item ) => {
-						return map.merge( item.cont );
-					}, new Map() );
-
-					// After getting all of the revisions, update the pages.
-					const pages = revisions.reduce( ( reduction, revision ) => {
-						if ( reduction.has( revision.pageid ) ) {
-							return reduction.setIn( [ revision.pageid, 'editors', revision.user ], true );
-						}
-
-						return reduction.set( revision.pageid, new Page( {
-							id: revision.pageid,
-							title: revision.title,
-							editors: new Map( {
-								[ revision.user ]: true
-							} )
-						} ) );
-					}, store.getState().pages )
-						// If the page is the same as what is already in the store,
-						// remove it from the map.
-						.filter( page => store.getState().pages.get( page.id ) !== page );
-
-					const pageSet = pages.reduce( ( set, page ) => {
-						const items = users.reduce( ( list, user ) => {
-							if ( page.editors.has( user ) ) {
-								return list;
-							}
-
-							// If the user is not one of the editors, add it to the list.
-							return list.add( {
-								page,
-								user
-							} );
-						}, new Set() );
+		.filter( () => store.getState().revisions.status === 'ready' )
+		.filter( () => store.getState().revisions.cont !== false )
+		.flatMap( () => {
+			// Set the variables so the request can be canceled if the state
+			// changes.
+			const wiki = store.getState().query.wiki;
+			const users = store.getState().query.user;
+			const startDate = store.getState().query.startDate;
+			const endDate = store.getState().query.endDate;
+			const cont = store.getState().revisions.cont;
 
-						return set.merge( items );
-					}, new Set() );
+			const request = Observable.ajax( {
+				url: buildRevisionUrl( wiki, users, startDate, endDate, cont ),
+				crossDomain: true,
+				responseType: 'json'
+			} )
+				.flatMap( ( ajaxResponse ) => {
```

Comment:
uhh... maybe both? on one hand a `switchMap` makes more sense since changing the query will issue a new fetch.. but what if you change the query but not everything is ready yet? (i.e. you remove one user but haven't added on back yet). The current request should probably be canceled immediately. but I think we can reduce the number with a `switchMap`